### PR TITLE
Fix `getUpdateComplete` example snippet and clean up formatting for others.

### LINKED ISSUE #1614: [reactive-element] Type change but not docs change for `getUpdateComplete`
https://github.com/Polymer/lit-html/blob/d649560cf7a20333dd1af6aa800fb36b40cc9c8f/packages/reactive-element/src/reactive-element.ts#L1088-L1106

This was previously `Promise<void>` as per the documented use in the method comment, but is now `Promise<boolean>`. It's not immediately clear to someone extended this functionality as per the documented use case what value should be returned in such a context.

### PR DESCRIPTION
**Pull request by bicknellr**
The `getUpdateComplete` example is wrong because you're now required to return `Promise<boolean>`. However, my understanding of how it's actually meant to be implemented is pretty poor. Should it consider anything else before returning the result of `super.getUpdateComplete()`? Or, is a proper implementation really meant to return that value without change at all times? I've assumed the second here.

Also Fixes #1614

### CODE REVIEW

File: packages/reactive-element/src/reactive-element.ts [hunk 1]
```diff
@@ -1168,11 +1168,15 @@ export abstract class ReactiveElement
    * ```js
    * class MyElement extends LitElement {
    *   async getUpdateComplete() {
```
Comment (AndrewJakubowicz) [inline packages/reactive-element/src/reactive-element.ts:1170]: Nit: override keyword?

**Review by AndrewJakubowicz — APPROVED** [review]
Amazing!

File: packages/reactive-element/src/reactive-element.ts (hunk 1 above)
Comment (bicknellr) [reply packages/reactive-element/src/reactive-element.ts:1170]: Added.

File: packages/reactive-element/src/reactive-element.ts [hunk 2]
```diff
@@ -1164,7 +1164,7 @@ export abstract class ReactiveElement
    * before fulfilling this Promise. To do this, first await
    * `super.getUpdateComplete()`, then any subsequent state.
    *
-   * @return A promise of a boolean that indicates if the update resolved
+   * @return A promise of a boolean that resolves to true if the update resolved
```
Comment (justinfagnani) [inline packages/reactive-element/src/reactive-element.ts:1151]:
```suggestion
   * @return A promise of a boolean that resolves to true if the update completed
```

File: packages/reactive-element/src/reactive-element.ts [hunk 3]
```diff
@@ -1191,6 +1191,8 @@ export abstract class ReactiveElement
    * }
    * ```
    *
+   * @return A promise of a boolean that resolves to true if the update resolved
+   *     without triggering another update.
```
Comment (justinfagnani) [inline packages/reactive-element/src/reactive-element.ts:1179]:
```suggestion
   * @return A promise of a boolean that resolves to true if the update completed
   *     without triggering another update.
```

**Review by justinfagnani — APPROVED** [review]
LGTM, but I'd remove one of the "resolved" instances per method.

**Pull request merged**